client/src/modules/cohort/index.jsx

```diff
@@ -27,18 +33,53 @@ class Cohort extends PureComponent {
     fetchListMetaData(id);
   }
 
+  showUpdateForm = () => {
+    this.setState({ updateFormVisibility: true });
+  };
+
+  hideUpdateForm = () => {
+    this.setState({ updateFormVisibility: false });
+  };
+
+  updateCohortHandler = cohortId => (name, description) => {
+    const { updateCohort } = this.props;
+    const dataToUpdate = {};
+    name ? (dataToUpdate.name = name) : null;
+    description ? (dataToUpdate.description = description) : null;
+
+    updateCohort(cohortId, dataToUpdate);
+    this.hideUpdateForm();
+  };
+
   render() {
-    const { cohortDetails, lists } = this.props;
+    const {
+      cohortDetails,
+      lists,
+      match: {
+        params: { id: cohortId }
+      }
+    } = this.props;
 
     if (!cohortDetails) {
       return null;
     }
 
     const { name, description } = cohortDetails;
-
+    const { updateFormVisibility } = this.state;
     return (
       <Fragment>
-        <Toolbar />
+        <Toolbar>
+          <ToolbarItem mainIcon={<EditIcon />} onClick={this.showUpdateForm} />
+        </Toolbar>
+        {updateFormVisibility && (
```

Comment:
proposal: pass updateFormVisibility as a e.g. isVisible prop to ModalForm and decide there  if creationForm should be render or not

Reply:
This is rebuilt in the coming implementation so I think it doesn't make any sense to change it now.